### ISSUE TITLE
When user uses keyboard to move to a field, if text is too large,it does not display fully

### ISSUE THREAD

**Issue by alon24** [carbon-design-system/carbon#7451]
In overflow menu, and in table when there is text (the text for the current menuitem, or table cell)  which does not display fully there is an accessibility issue.

When hovering over the text (overflow or table cell) the title is displayed as a tooltip, automaticaly.
BUT when user moves keyboard to the cell/menuitem (not mouse) then the text is never fully displayed
Based on this conversation`https://ibm-cloudplatform.slack.com/archives/C036P1CTN/p1607934465151500` I am opening this bug.

Open overflow menu, user with keyboard only is not able to see full text of links (this is in different langs which have different text, so the width does not hold the full text)

![image](https://user-images.githubusercontent.com/5415443/102090499-38597c00-3e26-11eb-8e1b-743d1c084b52.png)
![image](https://user-images.githubusercontent.com/5415443/102090520-3ee7f380-3e26-11eb-8d99-5bd85f2bc644.png)

**Comment by dakahn** [comment]
The ask here is to have any components that allow string truncation ('hi how are y....') to throw a custom Tooltip on focus (either via hover or keybaord) instead of relying on the `title` attribute.

This makes sense to me. Truncation as is seems problematic for any non-mouse user.

**Comment by alon24** [comment]
Any update? anyone?

**Comment by emyarod** [comment]
this will require a bit of research but here are a few related issues

https://github.com/carbon-design-system/carbon/issues/6605
https://github.com/carbon-design-system/carbon/issues/7088

**Comment by emyarod** [comment]
this will require a bit of research but here are a few related issues

https://github.com/carbon-design-system/carbon/issues/6605
https://github.com/carbon-design-system/carbon/issues/7088

**Comment by emyarod** [comment]
this will require a bit of research but here are a few related issues

https://github.com/carbon-design-system/carbon/issues/6605
https://github.com/carbon-design-system/carbon/issues/7088

cc @dakahn is allowing rich tooltips within overflow menu items alright now in the case of disabled items within the tooltip (in the issues linked above)?

**Comment by dakahn** [comment]
looking at my past comment [here](https://github.com/carbon-design-system/carbon/issues/6605#issuecomment-680989506)

i think that guidance is still right. if we stay away from native tooltips and instead use a dialog or a popover for this that triggers on focus (etc etc) we can make this work. could take a bit of prototyping size/shape/position (probably what you meant by bit of research). 

@carbon-design-system/design does a carbon tooltip triggered on focus of overflow menu items comply with spec for both tooltip and overflow menu? 

@joshblack is there anything I'm not thinking of here? I know we've talked about similar issues before regarding truncation and popovers.
